config/prometheus/kustomization.yaml

```diff
@@ -1,2 +1,3 @@
 resources:
+- rbac.yaml
```

Comment:
sorry I was not clear earlier: this should be gone, and so `config/prometheus/rbac.yaml`.
Now: this suggestion of mine is likely to break something. In that case, we need to know what breaks and when.

If the content of `rbac.yaml` is used just and only when we run `kustomize` in flows like `make deploy`, good, we can uncomment it again (in a separate commit preferably).

I'm insisting on this point because I want to be really sure we do NOT try to add stuff in the non-numaresources namespace in production flows.

For development flows (`make deploy` and the like) this is fine.

All the above is moot (and so we can merge this version) if we are fully confident the manifests are created just and only in the developer flows (and I can't verify this myself now)

Reply:
Well I'm pretty sure they are created only in the developer flows, and only when we unquote [this](https://github.com/openshift-kni/numaresources-operator/blob/main/config/default/kustomization.yaml#L25)  (will verify this of course!).
If I have unquoted this as part of this PR and then run `make generate bundle manifests` only then this will be part of the OLM installation.

Reply:
Anyhow, I need to figure out what will break ib case we delete this

Reply:
ok, this is the last concern, other than that LGTM. Once we are confident (preferably we can demonstrate) this affects only developer flows, we can merge. Otherwise we can undo this specific part and move to another PR.

Note all this discussion can possibly apply also to the RTE PR.

Reply:
The RTE PR is a bit different. we are not adding this role again or any other manifest except the service for the daemonset.
The role service account rolebinding and all that stuff should be declared applied only once and utilized by both RTE and the manager.
IOW, the RTE PR don't need to address most of this discussion comments (but there are other things maybe)

Reply:
@ffromani 
removing the _rbac.yaml_ will result in these errors in the Prometheus pods **when ServiceMonitor is applied** and instructs Prometheus pods to poll metrics:

```
\"system:serviceaccount:openshift-monitoring:prometheus-k8s\" cannot list resource \"pods\" in API group \"\" in the namespace \"numaresources\""
ts=2024-12-18T20:55:46.293Z caller=klog.go:118 level=warn component=k8s_client_runtime func=Warningf msg="github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:554: failed to list *v1.Endpoints: endpoints is forbidden: User \"system:serviceaccount:openshift-monitoring:prometheus-k8s\" cannot list resource \"endpoints\" in API group \"\" in the namespace \"numaresources\""
ts=2024-12-18T20:55:48.126Z caller=klog.go:118 level=warn component=k8s_client_runtime func=Warningf msg="github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:555: failed to list *v1.Service: services is forbidden: User \"system:serviceaccount:openshift-monitoring:prometheus-k8s\" cannot list resource \"services\" in API group \"\" in the namespace \"numaresources\""

```

But again we are not applying this by default, neither on developer and official installation flows.

Reply:
ok, this needs to be investigated further